resource/name.go

```diff
@@ -51,7 +51,7 @@ func newRemoteName(remoteName string, api API, name string) Name {
 // NewFromString creates a new Name based on a fully qualified resource name string passed in.
 func NewFromString(resourceName string) (Name, error) {
 	if !resRegexValidator.MatchString(resourceName) {
-		return Name{}, errors.Errorf("string %q is not a valid resource name", resourceName)
+		return Name{}, errors.Errorf("string %q is not a fully qualified resource name", resourceName)
```

Comment:
`NewFromString` is used [here](https://github.com/viamrobotics/rdk/blob/main/resource/resource_graph.go#L548) to try to resolve dependencies; an error might not indicate an "invalid" resource name, but rather that the name should be thereon treated as "short" or "simple", and we should call `findNodesByShortName`. `robot1:robot2:robot3:e` was being treated as a valid, fully qualified resource name because it `MatchString`ed above; with the change to the regex, it is now correctly seen as "not a fully qualified resource name".